Simplenote/CoreDataManager.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+import CoreData
+
+@objcMembers
+class CoreDataManager: NSObject {
+
+    /// Storage Settings
+    ///
+    let storageSettings: StorageSettings
+
+    @objc
+    init(storageSettings: StorageSettings) {
+        self.storageSettings = storageSettings
+        super.init()
+    }
+
+    // MARK: Core Data
+    private(set) lazy var managedObjectModel: NSManagedObjectModel = {
+        guard let mom = NSManagedObjectModel(contentsOf: storageSettings.modelURL) else {
+            fatalError()
+        }
+        return mom
+    }()
+
+    private(set) lazy var managedObjectContext: NSManagedObjectContext = {
+        let moc = NSManagedObjectContext(concurrencyType: .mainQueueConcurrencyType)
+        moc.undoManager = nil
+        return moc
+    }()
+
+    private(set) lazy var persistentStoreCoordinator: NSPersistentStoreCoordinator = {
+        let psc = NSPersistentStoreCoordinator(managedObjectModel: managedObjectModel)
+
+        let options = [NSMigratePersistentStoresAutomaticallyOption: true, NSInferMappingModelAutomaticallyOption: true, NSSQLitePragmasOption: [Constants.journalMode: Constants.journalSetting]] as [AnyHashable: Any]
+
+        // Testing logs
+        //
+        NSLog("🎯 Loading PersistentStore at URL: \(storageSettings.storageURL)")
+
+        do {
+            try psc.addPersistentStore(ofType: NSSQLiteStoreType, configurationName: nil, at: storageSettings.storageURL, options: options)
+        } catch {
+            NSLog("Unresolved Error")
+        }
```

Comment:
This should just crash, there is no way we can use the app without core data :-)

Reply:
+1 `fatalError()` please 😄 

It's one of those errors that, if happens, you're better off tearing down the whole app.

Reply:
Fixed this by adding CrashLogging.crash() which will remote log the issue and terminate the app